e2e/cat_test.go

```diff
@@ -305,4 +322,215 @@ func TestCatByVersionID(t *testing.T) {
 			t.Errorf("(-want +got):\n%v", diff)
 		}
 	}
+
+	version := "1"
+
+	// wildcard and prefix fail cases
+	cmd = s5cmd("cat", "--version-id", version, "s3://"+bucket+"/")
+	result = icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+	assertLines(t, result.Stderr(), map[int]compareFunc{
+		0: equals(`ERROR "cat --version-id=%v s3://%v/": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+
+	cmd = s5cmd("cat", "--version-id", version, "s3://"+bucket+"/folder/")
+	result = icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+	assertLines(t, result.Stderr(), map[int]compareFunc{
+		0: equals(`ERROR "cat --version-id=%v s3://%v/folder/": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+
+	cmd = s5cmd("cat", "--version-id", version, "s3://"+bucket+"/*")
+	result = icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+	assertLines(t, result.Stderr(), map[int]compareFunc{
+		0: equals(`ERROR "cat --version-id=%v s3://%v/*": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+}
+
+func TestCatPrefix(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		files    []string
+		contents []string
+		prefix   string
+		expected string
+	}{
+		{
+			files:    []string{"file1.txt", "file2.txt"},
+			contents: []string{"content0", "content1"},
+			expected: "content0content1"},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt"},
+			contents: []string{"content0", "content1", "content2", "content3"},
+			expected: "content0content1",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt"},
+			contents: []string{"content0", "content1", "content2", "content3"},
+			prefix:   "dir/",
+			expected: "content2content3",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt", "dir/nesteddir/file5.txt"},
+			contents: []string{"content0", "content1", "content2", "content3", "content4"},
+			prefix:   "dir/",
+			expected: "content2content3",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt", "dir/nesteddir/file5.txt"},
+			contents: []string{"content0", "content1", "content2", "content3", "content4"},
+			prefix:   "dir/nesteddir/",
+			expected: "content4",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.expected, func(t *testing.T) {
```

Comment:
instead of expected, please give a name to your subtests, example;
https://github.com/peak/s5cmd/blob/master/storage/url/url_test.go#L16-L21

if not, just use iteration index of the loop.

Reply:
Sure, named them case-[idx].